src/views/OtherStoresInventoryModal.vue

```diff
@@ -58,38 +122,201 @@ export default defineComponent({
+      if (!this.facilityLatLon(currentFacilityId).latitude || !this.facilityLatLon(currentFacilityId).longitude) {
+        // Fetching Lat Lon for current facility
+        await this.store.dispatch("util/fetchCurrentFacilityLatLon", currentFacilityId);
+      }
+
+      if (!this.storesInformation || this.storesInformation.length === 0) {
+        // Fetching store lookup
+        if (this.facilityLatLon(currentFacilityId)?.latitude && this.facilityLatLon(currentFacilityId)?.longitude) {
+          await this.store.dispatch("util/fetchStoresInformation", {
+            latitude: this.facilityLatLon(currentFacilityId).latitude,
+            longitude: this.facilityLatLon(currentFacilityId).longitude
+          });
+        }
+      }
+
+      // Map the stock information
+      this.storesWithInventory = this.mapStock(this.storesInventory, this.storesInformation);
+    } catch (error) {
+      // Fallback mapping when APIs fail
+      this.storesWithInventory = this.storesInventory.map((facility: any) => ({
+        storeCode: facility.facilityId,
+        storeName: facility.facilityName,
+        stock: facility.stock
+      }));
+    } finally {
+      this.isRefreshing = false;
+    }
+
+    // Sort by name by default
+    this.sortByName();
   },
   methods: {
     closeModal() {
       modalController.dismiss({ dismissed: true });
     },
-    searchFacilities(){
+    searchFacilities() {
+      this.isRefreshing = true;
+      let filteredInventory = this.mapStock(this.storesInventory, this.storesInformation);
       if (this.queryString.trim() !== "") {
-        this.storesInventory = this.otherStoresInventory.filter((facility: any) => facility.facilityName.toLowerCase().includes(this.queryString.toLowerCase()));
+        filteredInventory = filteredInventory.filter((store: any) =>
+          store.storeName.toLowerCase().includes(this.queryString.toLowerCase())
+        );
+      }
+      if (this.hideEmptyStores) {
+        filteredInventory = filteredInventory.filter((store: any) => store.stock > 0);
+      }
+      this.storesWithInventory = filteredInventory;
+      this.isRefreshing = false;
+    },
+    // method to map the facility inventory from props to the store lookup data from api call
+    mapStock(storesInventory: any[], storesInformation: any[]): any[] {
+      return storesInformation.slice(1).map((store, index) => {  // skips first item (current facility)
+        const matchingFacility = storesInventory.find(facility => facility.facilityId === store.storeCode);
+        const storeHoursInfo = this.getStoreHoursStatus(store);
+        return {
+          ...store,
+          index: index + 1,
+          stock: matchingFacility ? matchingFacility.stock : 0,
+          hoursDisplay: storeHoursInfo.display,
+          isOpen: storeHoursInfo.isOpen
+        };
+      });
+    },
+    // logic for sorting stores by names
+    sortByNameLogic(a: any, b: any): number {
```

Comment:
We should think on improving this method name

Reply:
as this logic is for sorting stores alphabetically by names,
should i call it `sortStoresAlphabetically` ?